pyfstat/make_sfts.py

```diff
@@ -413,6 +413,170 @@ def predict_fstat(self):
         return twoF_expected
 
 
+class BinaryModulatedWriter(Writer):
+    """ Instance object for generating SFTs containing binary continuous signals """
+
+    @helper_functions.initializer
+    def __init__(
+        self,
+        label="Test",
+        tstart=700000000,
+        duration=100 * 86400,
+        tref=None,
+        F0=30,
+        F1=1e-10,
+        F2=0,
+        Alpha=5e-3,
+        Delta=6e-2,
+        orbitTp=0.0,
+        orbitArgp=0.0,
+        orbitasini=0.0,
+        orbitEcc=0.0,
+        orbitPeriod=0.0,
+        h0=0.1,
+        cosi=0.0,
+        psi=0.0,
+        phi=0,
+        Tsft=1800,
+        outdir=".",
+        sqrtSX=1,
+        Band=4,
+        detectors="H1",
+        minStartTime=None,
+        maxStartTime=None,
+        add_noise=True,
+        transientWindowType="none",
+    ):
+        """
+        Parameters
+        ----------
+        label: string
+            a human-readable label to be used in naming the output files
+        tstart, duration : int
+            start and duration (in gps seconds) of the total observation span
+        tref: float or None
+            reference time (default is None, which sets the reference time to
+            tstart)
+        F0, F1, F2, Alpha, Delta, orbitTp, orbitArgp, orbitasini, orbitEcc, h0, cosi, psi, phi: float
+            frequency, sky-position, binary orbit and amplitude parameters
+        Tsft: float
+            the sft duration
+        minStartTime, maxStartTime: float
+            if not None, the total span of data, this can be used to generate
+            transient signals
+        see `lalapps_Makefakedata_v5 --help` for help with the other paramaters
+        """
+        super().__init__(
+            label,
+            tstart,
+            duration,
+            tref,
+            F0,
+            F1,
+            F2,
+            Alpha,
+            Delta,
+            h0,
+            cosi,
+            psi,
+            phi,
+            Tsft,
+            outdir,
+            sqrtSX,
+            Band,
+            detectors,
+            minStartTime,
+            maxStartTime,
+            add_noise,
+            transientWindowType,
+        )
+
+        self.parse_args_consistent_with_mfd()
+
+    def parse_args_consistent_with_mfd(self):
+        """ This will allow us to get rid of the get_single_config_* family
+            Future rework may use a dictionary as the default input method,
```

Comment:
This interface looks indeed a bit nicer than the one on the original Writer class! So eventually one could merge the two based on this new one. But for now I'm fine adding this one and leaving the original as is.